.github/workflows/ci.yml

```
@@ -0,0 +1,121 @@
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+## events but only for the master branch
+on:
+  push:
+    branches:
+      - 'develop'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }} 
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install apt packages
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get install -qq \
+            libxml2-dev libxslt1-dev gfortran libatlas-base-dev \
+            libespeak1 libxcb-image0 libxcb-keysyms1 libxcb-render-util0 \
+            libxkbcommon-x11-0 libxcb-icccm4 libxcb1 openssl \
+            libxcb-randr0-dev libxcb-xtest0-dev libxcb-xinerama0-dev \
+            libxcb-shape0-dev libxcb-xkb-dev xvfb
+
+      - name: Download AppImage tool
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          wget \
+            https://github.com/AppImage/AppImageKit/releases/download/12/appimagetool-x86_64.AppImage \
+            -O $HOME/appimagetool-x86_64.AppImage
+          chmod a+x $HOME/appimagetool-x86_64.AppImage
+
+      - name: Set cache dir
+        run: echo "pip_cache_dir=$(pip cache dir)" >> $GITHUB_ENV
+        shell: bash
+        
+      - name: Fetch Cache
+        id: cache-target
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pip_cache_dir }}
+          key: ${{ runner.os }}-${{ matrix.python-version }}
+
+      - name: Set up environment
+        run: |
+          # Update pip.
+          python -m pip install -U pip setuptools wheel
+
+          # Install dependencies for tests.
+          pip install --progress-bar=off -U -r tests/requirements-tools.txt -r tests/requirements-libraries.txt
+
+          # Install PyInstaller Hook Sample, to ensure that tests declared in
+          # entry-points are discovered.
+          pip install https://github.com/pyinstaller/hooksample/archive/v4.0rc1.zip
+
+          # Compile bootloader
+          cd bootloader
+          python waf distclean all
```

`distclean` is included as part of `all`.
```suggestion
          python waf all
```